### PR TITLE
Add DAOWizardlet

### PR DESCRIPTION
**Pull request by KernelDeimos**
This PR adds a wizardlet which puts its data into the corresponding DAO (or one specified by a `daoKey`).

The following example, which can be run in the Javascript console, creates a wizard for adding Cron jobs:

```
ctrl.add(foam.u2.dialog.Popup.create({ closeable: false }, ctrl.__subContext__).tag({
  class: 'foam.u2.wizard.StepWizardletView',
  data: foam.u2.wizard.StepWizardletController.create({
    wizardlets: [ foam.u2.wizard.DAOWizardlet.create({
      title: 'Cron Wizard',
      of: 'foam.nanos.cron.Cron',
    }, ctrl.__subContext__) ]
  }),
  onClose: (x) => {
    x.closeDialog();
  }
}));
```

(note that `ctrl.__subContext__` arguments are only needed if you run this from the javascript console; in a regular view it should work as expected without those)

### CODE REVIEW

**Comment by KernelDeimos** [comment]
~This maybe works fine, but right now if it's used in StepWizardletView the submit button doesn't work. I'm currently investigating the cause of this.~ I was testing it wrong

File: src/foam/u2/wizard/DAOWizardlet.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
+foam.CLASS({
```
Comment (kgrgreer) [inline src/foam/u2/wizard/DAOWizardlet.js:7]: copyright

Comment (KernelDeimos) [reply src/foam/u2/wizard/DAOWizardlet.js:7]: One day I'll submit a PR with all the license headers already added. Maybe not today, but one day

Comment (KernelDeimos) [reply src/foam/u2/wizard/DAOWizardlet.js:7]: It's fixed on my branch but Github isn't updating this PR for some reason

**Comment by KernelDeimos** [comment]
...

**Pull request closed without merging**